src/transport_ofi.c

```diff
@@ -99,28 +99,124 @@ static inline void init_dt_size(void)
   SHMEM_Dtsize[FI_LONG_DOUBLE_COMPLEX] = sizeof(long double complex);
 }
 
-static int SHM_DT_INT[]=
+static char * SHMEM_DtName[FI_DATATYPE_LAST]=
 {
-  SHM_INTERNAL_SHORT, SHM_INTERNAL_INT, SHM_INTERNAL_LONG,
+  "int8", "uint8", "int16", "uint16", "int32", "uint32", "int64", "uint64"
+  "float", "double", "float complex", "double complex", "long double",
+  "long double complex"
```

Comment:
Need to be sure that we can rely on the values of the FI_* type and op constants landing on these array indices.